component/http/cache/cache_model.go

```diff
@@ -0,0 +1,68 @@
+package cache
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// handlerRequest is the dedicated request object for the cache handler
+type handlerRequest struct {
+	header string
+	path   string
+	query  string
+}
+
+// toCacheHandlerRequest transforms the http Request object to the cache handler request
+func toCacheHandlerRequest(req *http.Request) *handlerRequest {
+	var header string
+	if req.Header != nil {
+		header = req.Header.Get(HeaderCacheControl)
+	}
+	var path string
+	var query string
+	if req.URL != nil {
+		path = req.URL.Path
+		query = req.URL.RawQuery
+	}
+	return &handlerRequest{
+		header: header,
+		path:   path,
+		query:  query,
+	}
+}
+
+// getKey generates a unique cache key based on the route path and the query parameters
+func (c *handlerRequest) getKey() string {
+	return fmt.Sprintf("%s:%s", c.path, c.query)
+}
+
+// handlerResponse is the dedicated Response object for the cache handler
+type handlerResponse struct {
+	Bytes  []byte
+	Header http.Header
+}
+
+// cachedResponse is the struct representing an object retrieved or ready to be put into the route cache
+type cachedResponse struct {
```

Comment:
since we are in the cache package could we name if just:
```suggestion
type response struct {
```